sqlx-core/src/decode.rs

```diff
@@ -35,3 +33,16 @@ where
         }
     }
 }
+
+pub trait TypedDecode<'q, DB>: Decode<'q, DB> + Sized + Type<DB>
```

Comment:
This is just a shortcut for `Decode + Type`

Reply:
I'd rather not increase the API surface at this time without a strong reason. If we provide `TypedDecode` why not `DecodeOwned` and `TypedDecodeOwned`, etc. 

Orthogonally it may make a lot of sense to make the `Type` trait be `Type: Decode + Encode`. 

---

On the other side here, where are you using `Decode + Type` ? That was a thing in v0.3 but in master, `Row::try_get` should now be `T::accepts(&value.type_info())` instead of `T::type_info().compatible(&value.type_info())`.

Reply:
There were places where `T: Decode/Encode` was assumed to also have `Type` and made explicitly use of one of its methods.

https://github.com/launchbadge/sqlx/blob/84d7f9f21d1b695e721326c60b2003e0e9f81ee9/sqlx-core/src/encode.rs#L69-L77

In the above example,  `T: 'q + TypedEncode<'q, DB>` was used instead of `T: 'q + Encode<'q, DB> + Type<DB>`. I didn't modify any logic

Reply:
No problem, I will remove these new traits